modules/parsers/parser-hdbtable/com.sap.xsk.parser.hdbtable/src/main/java/com/sap/xsk/parser/hdbtable/custom/XSKHDBTABLECoreVisitor.java

```diff
@@ -17,229 +17,298 @@
+                hdbtableObject.add(HDBTABLE_PROPERTY_TABLE_TYPE, visitTableTypeProp((HdbtableParser.TableTypePropContext) tree.getChild(0)));
+            }else if (tree.getChild(0) instanceof HdbtableParser.TableColumnsPropContext) {
+                checkPropertyDeclarationUniqueness(HDBTABLE_PROPERTY_COLUMNS, uniqueTableProperties);
+                hdbtableObject.add (HDBTABLE_PROPERTY_COLUMNS, visitTableColumnsProp((HdbtableParser.TableColumnsPropContext) tree.getChild(0)));
+            }else if (tree.getChild(0) instanceof HdbtableParser.TableIndexesPropContext){
+                checkPropertyDeclarationUniqueness(HDBTABLE_PROPERTY_INDEXES, uniqueTableProperties);
+                hdbtableObject.add (HDBTABLE_PROPERTY_INDEXES, visitTableIndexesProp((HdbtableParser.TableIndexesPropContext) tree.getChild(0)));
+            }else if (tree.getChild(0) instanceof HdbtableParser.TablePrimaryKeyPropContext){
+                checkPropertyDeclarationUniqueness(HDBTABLE_PROPERTY_PKCOLUMNS, uniqueTableProperties);
+                hdbtableObject.add(HDBTABLE_PROPERTY_PKCOLUMNS, visitTablePrimaryKeyProp((HdbtableParser.TablePrimaryKeyPropContext) tree.getChild(0)));
+            }else if (tree.getChild(0) instanceof HdbtableParser.DescriptionPropContext){
+                checkPropertyDeclarationUniqueness(HDBTABLE_PROPERTY_DESCRIPTION, uniqueTableProperties);
+                hdbtableObject.add(HDBTABLE_PROPERTY_DESCRIPTION, visitDescriptionProp((HdbtableParser.DescriptionPropContext) tree.getChild(0)));
+            }else if(tree.getChild(0) instanceof HdbtableParser.TemporaryPropContext) {
+                checkPropertyDeclarationUniqueness(HDBTABLE_PROPERTY_TEMPORARY, uniqueTableProperties);
+                hdbtableObject.add(HDBTABLE_PROPERTY_TEMPORARY, visitTemporaryProp((HdbtableParser.TemporaryPropContext) tree.getChild(0)));
+            }else if(tree.getChild(0) instanceof HdbtableParser.PublicPropContext) {
+                checkPropertyDeclarationUniqueness(HDBTABLE_PROPERTY_PUBLIC, uniqueTableProperties);
+                hdbtableObject.add(HDBTABLE_PROPERTY_PUBLIC, visitPublicProp((HdbtableParser.PublicPropContext) tree.getChild(0)));
+            }else if(tree.getChild(0) instanceof HdbtableParser.LoggingTypePropContext) {
+                checkPropertyDeclarationUniqueness(HDBTABLE_PROPERTY_LOGGING_TYPE, uniqueTableProperties);
+                hdbtableObject.add (HDBTABLE_PROPERTY_LOGGING_TYPE, visitLoggingTypeProp((HdbtableParser.LoggingTypePropContext) tree.getChild(0)));
+            }else if(tree.getChild(0) instanceof HdbtableParser.TablePrimaryKeyIndexTypePropContext) {
+                checkPropertyDeclarationUniqueness(HDBTABLE_PROPERTY_INDEX_TYPE, uniqueTableProperties);
+                hdbtableObject.add(HDBTABLE_PROPERTY_INDEX_TYPE, visitTablePrimaryKeyIndexTypeProp((HdbtableParser.TablePrimaryKeyIndexTypePropContext) tree.getChild(0)));
+            }
```

Comment:
Can the getChild(0) be replaced by ctx.schemaNameProp() and skip the check for 'instanceof HdbtableParser.SchemaNamePropContex?

Can look like 
if(ctx.schemaNameProp()!null){ do the logic}
The same comment for the other else branches.